contracts/wveOLAS.sol

```diff
@@ -0,0 +1,170 @@
+    function getPastVotes(address account, uint256 blockNumber) external view returns (uint256 balance) {
+        // Get the zero account point
+        PointVoting memory uPoint = getUserPoint(account, 0);
+        // Check that the point exists and the zero point block number is not smaller than the specified blockNumber
+        if (uPoint.blockNumber > 0 && blockNumber >= uPoint.blockNumber) {
+            balance = IVEOLAS(ve).getPastVotes(account, blockNumber);
+        }
+    }
+
+    /// @dev Gets the account balance at a specific block number.
+    /// @param account Account address.
+    /// @param blockNumber Block number.
+    /// @return balance Account balance.
+    function balanceOfAt(address account, uint256 blockNumber) external view returns (uint256 balance) {
+        // Get the zero account point
+        PointVoting memory uPoint = getUserPoint(account, 0);
+        // Check that the zero point block number is not smaller than the specified blockNumber
+        if (blockNumber >= uPoint.blockNumber) {
+            balance = IVEOLAS(ve).balanceOfAt(account, blockNumber);
+        }
+    }
+
+    /// @dev Gets total token supply at a specific block number.
+    /// @param blockNumber Block number.
+    /// @return supplyAt Supply at the specified block number.
+    function totalSupplyAt(uint256 blockNumber) external view returns (uint256 supplyAt) {
+        supplyAt = IVEOLAS(ve).totalSupplyAt(blockNumber);
+    }
+
+    /// @dev Calculates total voting power at time `ts` that must be greater than the last supply point timestamp.
+    /// @param ts Time to get total voting power at.
+    /// @return vPower Total voting power.
+    function totalSupplyLockedAtT(uint256 ts) external view returns (uint256 vPower) {
+        // Get the total number of supply points
+        uint256 numPoints = IVEOLAS(ve).totalNumPoints();
+        PointVoting memory sPoint = IVEOLAS(ve).mapSupplyPoints(numPoints);
+        // Check the last supply point timestamp is not smaller than the specified ts
+        if (ts >= sPoint.ts) {
+            vPower = IVEOLAS(ve).totalSupplyLockedAtT(ts);
+        } else {
+            revert WrongTimestamp(sPoint.ts, ts);
+        }
+    }
+
+    /// @dev Calculate total voting power at some point in the past.
+    /// @notice The requested block number must be at least equal to the zero supply point block number.
+    /// @param blockNumber Block number to calculate the total voting power at.
+    /// @return vPower Total voting power.
+    function getPastTotalSupply(uint256 blockNumber) external view returns (uint256 vPower) {
+        // Get the zero supply point
+        PointVoting memory sPoint = IVEOLAS(ve).mapSupplyPoints(0);
+        // Check the requested block number to be at least equal to the zero supply point block number
+        if (blockNumber >= sPoint.blockNumber) {
+            vPower = IVEOLAS(ve).getPastTotalSupply(blockNumber);
+        }
```

Comment:
If the `blockNumber` is not smaller than the very first supply point (e.g. the zero supply point), the `getPastTotalSupply` veOLAS original behavior is correct. So the latter is called.
Otherwise, we return zero instead reverting without mentioning the reasons ( changing the native behavior of the original Curve Voting Escrow that plain reverts in this case)